src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -478,8 +484,13 @@ private void UpdateTopologyCallback(object _)
 
             _lastKnownUrls = initialUrls;
 
-            throw new AggregateException("Failed to retrieve cluster topology from all known nodes" + Environment.NewLine +
-                                         string.Join(Environment.NewLine, list.Select(x => x.Item1 + " -> " + x.Item2?.Message))
+            ThrowExceptions(list);
+        }
+
+        protected virtual void ThrowExceptions(List<(string, Exception)> list)
+        {
```

Comment:
Leaving my comment from previous PR: https://github.com/ravendb/ravendb/pull/4356#discussion_r152026309 for further discussion.